skbio/stats/composition.py

```diff
@@ -608,10 +611,269 @@ def centralize(mat):
 
     """
     mat = closure(mat)
-    cen = ss.gmean(mat, axis=0)
+    cen = scipy.stats.gmean(mat, axis=0)
     return perturb_inv(mat, cen)
 
 
+@experimental(as_of="0.4.0-dev")
+def ancom(table, groups,
+          alpha=0.05,
+          tau=0.02,
+          theta=0.1,
+          multiple_comparisons_correction=None,
+          significance_test=scipy.stats.ttest_ind):
+    r""" Performs a differential abundance test using ANCOM
+
+    This is done by calculating pairwise log ratios between all features
+    and performing a signficance test to determine if there is a
+    significant difference in feature ratios with respect to the
+    variable of interest.
+
+    In an experiment with only two treatments, this test tests the
+    following hypothesis for feature :math:`i`
+
+    .. math::
+        H_{0i}:\mathbb{E}[\ln(u_i^{(1)})] = \mathbb{E}[\ln(u_i^{(2)})]
+
+    where :math:`u_i^{(1)}` is the mean abundance for feature
+    :math:`i` in the first group and :math:`u_i^{(2)}` is the
+    mean abundance for feature :math:`i` in the second group.
+
+    This method can be extended to an arbitrary number of classes
+    by passing in a different statistical function.
+
+    Parameters
+    ----------
+    table : pd.DataFrame or array_like
+       A 2D matrix of counts or proportions where
+       the rows correspond to samples and the
+       columns correspond to features
+    groups : pd.Series or 1D array_like
+       Vector of group categories
+    alpha : float, optional
+       Significance level for each of the statistical tests
+    tau : float, optional
+       A constant used to determine an appropriate
+       cutoff
+    theta : float, optional
+       Lower bound for the proportion of differential features
+    multiple_comparisons_correction: {None, 'holm-bonferroni'}, optional
+       The multiple comparison correction procedure to run
+    significance_test: function, optional
+       A statistical signficance function to test for
+       signficance between classes.
+
+    Returns
+    -------
+    pd.DataFrame
+        A table of features, their W-statistics and whether
+        the null hypothesis is rejected
+
+    References
+    ----------
+    ..[1] Mandal et al. "Analysis of composition of microbiomes:
+          a novel method for studying microbial composition",
+          Microbial Ecology in Health & Disease, (2015), 26
+
+    Examples
+    --------
+    First import all of the necessary modules
+
+    >>> from skbio.stats.composition import ancom
+    >>> import pandas as pd
+
+    Now lets load in a pandas dataframe with sample and feature ids
+    for our data matrix
+
+    >>> table = pd.DataFrame(
+    ...     [[10., 11., 10., 10., 10., 10., 10.],
```

Comment:
Since this method was originally intended for proportions (and because the method is in `stats.composition`), it probably makes sense to use that type of data here to show off the method's typical usage.